packages/create-cloudflare/src/frameworks/next/index.ts

```diff
@@ -72,16 +73,74 @@ const configure = async (ctx: PagesGeneratorContext) => {
 	writeFile(handlerPath, handlerFile);
 	updateStatus("Created an example API route handler");
 
+	const installEslintPlugin = await shouldInstallNextOnPagesEslintPlugin(ctx);
+
+	if (installEslintPlugin) {
+		await writeEslintrc(ctx);
+		updateStatus("eslint-plugin-next-on-pages added to eslintrc");
+	}
+
 	// Add some dev dependencies
 	process.chdir(projectName);
-	const packages = ["@cloudflare/next-on-pages@1", "vercel"];
+	const packages = [
+		"@cloudflare/next-on-pages@1",
+		"vercel",
+		...(installEslintPlugin ? ["eslint-plugin-next-on-pages"] : []),
+	];
 	await installPackages(packages, {
 		dev: true,
 		startText: "Adding the Cloudflare Pages adapter",
 		doneText: `${brandColor(`installed`)} ${dim(packages.join(", "))}`,
 	});
 };
 
+// Note: this isn't a generic helper since it's Next specific, this can be
+//       made into a generic helper by checking all possible types of configs
+//       (including a field in the package.json)
+//       (and potentially returning what type of config is used)
+export const usesEslint = (projectRoot = ".") => {
+	// TODO: check other types of config files, etc...
+	return existsSync(`${projectRoot}/.eslintrc.json`);
+};
+
+export const shouldInstallNextOnPagesEslintPlugin = async (
+	ctx: PagesGeneratorContext
+): Promise<boolean> => {
+	if (!usesEslint(ctx.project.name)) return false;
+
+	// TODO: ask james about this
+	// I want to ask the question but I think we should also allow users to
+	// provide this via a flag in case they don't want any interactivity at all
+	return await processArgument(ctx.args, "eslint-plugin" as keyof C3Args, {
```

Comment:
I want to ask the question but I think we should also allow users to provide this via a flag in case they don't want any interactivity at all, is this right/correct?